src/Uno.UWP/Extensions/PermissionHelper.Android.cs

```diff
@@ -0,0 +1,198 @@
+							allPermissions.Add(permission);
+					}
+				}
+			}
+
+			// prepare list of permission to ask for
+			var askForPermission = new List<string>();
+
+			// check if permission is granted
+			foreach (var permission in allPermissions)
+			{
+				if (Android.Support.V4.Content.ContextCompat.CheckSelfPermission(context, permission)
+						!= Android.Content.PM.Permission.Granted)
+				{
+					askForPermission.Add(permission);
+				}
+			}
+
+			return askForPermission.ToArray();
+
+		}
+
+
+		/// <summary>
+		/// Return true if requiredPermission is granted. Show dialog asking for missing permission.
+		/// </summary>
+		/// <param name="requiredPermission">Permission that are required</param>
+		/// <returns>Bool if permission is granted, false if not</returns>
+		public static Task<bool> AndroidPermissionAsync(string requiredPermission)
+			=> AndroidPermissionAsync(new string[] { requiredPermission }, null);
+
+		/// <summary>
+		/// Return true if granted are: requiredPermission, and optionalPermissions (if mentioned in Manifest). Show dialog asking for missing permissions..
+		/// </summary>
+		/// <param name="requiredPermissions">Permission that are required</param>
+		/// <param name="optionalPermissions">Permission that are required only if declared in Manifest</param>
+		/// <returns>Bool if all permissions are granted, false if any of them is not granted</returns>
+		public static Task<bool> AndroidPermissionAsync(string requiredPermission, string optionalPermission)
+			=> AndroidPermissionAsync(new string[] { requiredPermission }, new string[] { optionalPermission });
+
+
+		/// <summary>
+		/// Return true if granted are: all requiredPermissions, and all optionalPermissions mentioned in Manifest. Show dialog asking for missing permissions. Both parameters can be null.
+		/// </summary>
+		/// <param name="requiredPermissions">Permissions that are required. Can be null.</param>
+		/// <param name="optionalPermissions">Permissions that are required only if declared in Manifest. Can be null.</param>
+		/// <param name="ignoreErrors">if 'true', then do not throw exception on error, and simply retur false</param>
+		/// <returns>Bool if all permissions are granted, false if any of them is not granted</returns>
+		public static Task<bool> AndroidPermissionAsync(string[] requiredPermissions, string[] optionalPermissions, bool ignoreErrors = false)
+		{
+			// prepare return value
+			_permissionCompletionSource = new TaskCompletionSource<bool>();
+
+			var askForPermission = MissingPermissions(requiredPermissions, optionalPermissions);
+			if (askForPermission is null)
+			{
+				if (ignoreErrors)
+				{
+					_permissionCompletionSource.SetResult(false); // signal: "permission denied", although there is some error
+					return _permissionCompletionSource.Task;
+				}
+				throw new AccessViolationException("MissingPermissions returned ERROR - check Manifest file");

```

Comment:
Don't use `AccessViolationException`, this is reserved for native generated exceptions.

Reply:
Which Exception throw instead?

Reply:
InvalidOperationException will do.